### PR TITLE
Added support for allowing image  pull from local registry

### PR DESCRIPTION
**Pull request by panktishah26**
## Description

Allow fetching images from local registries in place of public registries. 

## Why is this needed
We cannot use images from local registries as hook bootkit doesn't support local registry certificate as well as insecure pull from local registries. With this PR, I have added insecure registry support for hook bootkit.

Fixes: #

## How Has This Been Tested?
I have tested these changes by creating EKS-A baremetal cluster by giving local registry URL as well as public one.

### CODE REVIEW

File: hook-docker/main.go [hunk 1]
```diff
@@ -59,8 +60,19 @@ func main() {
 		LogOpts: map[string]string{
 			"syslog-address": fmt.Sprintf("udp://%v:514", cfg.syslogHost),
 		},
+		Insecure_registries: cfg.insecure_registries,
 	}
-	if err := d.writeToDisk("/etc/docker/daemon.json"); err != nil {
+
+	path = "/etc/docker/"
+
+	err = os.MkdirAll(path, os.ModeDir)
```
Comment (pokearu) [inline hook-docker/main.go:68]: Any reason we needed the directory created separately? Did just calling `writeToDisk` not work?

File: hook-docker/main.go [hunk 2]
```diff
@@ -17,14 +17,15 @@ type tinkConfig struct {
...
-
+	insecure_registries []string
 	// TODO add others
 }
 
 type dockerConfig struct {
 	Debug     bool              `json:"debug"`
 	LogDriver string            `json:"log-driver,omitempty"`
 	LogOpts   map[string]string `json:"log-opts,omitempty"`
+	Insecure_registries []string	`json:"insecure-registries,omitempty"`
```
Comment (pokearu) [inline hook-docker/main.go:28]: nit: can we run a linter? The struct alignment looks a bit off.

File: hook-docker/main.go [hunk 3]
```diff
@@ -59,8 +60,19 @@ func main() {
...
-	if err := d.writeToDisk("/etc/docker/daemon.json"); err != nil {
+
+	path = "/etc/docker/"
+
+	err = os.MkdirAll(path, os.ModeDir)
+	if err != nil {
+		panic(err)
+	}
+
+	daemonPath := fmt.Sprintf("%sdaemon.json", path)
```
Comment (jacobweinstock) [inline hook-docker/main.go:73]: might think about using `path.Join` here. https://pkg.go.dev/path#example-Join

**Comment by jacobweinstock** [comment]
This functionality now exist in main via https://github.com/tinkerbell/hook/pull/150

**Pull request closed without merging**